primer/brand
# improvements to button appearance and documentation

## Summary

Multiple improvements to the Buttons visual appearance, and its documentation in Storybook (with more visual testing coverage applied as a result).

## List of notable changes:



- **added** an inactive visual appearance to the `Button`, which it previously didn't have
- **added** additional Storybook stories for: hover, focus and disabled states
- **added** visual screenshot coverage of all the above
- **added** new design tokens to apply required changes and enable app-side customisation

- **updated** Storybook structure to match recent ADR around playgrounds, defaults and feature
- **updated** focus styles from previously bespoke grey to global focus tokens

## What should reviewers focus on?

- Try out Storybook and see if the states work as expected

## Steps to test:


1. Open the Button in Storybook with [this link](#).
2. Cycle through the stories and playground
3. Verify that all Storybook controls behaves as expected

## Supporting resources (related issues, external links, etc):

- [Designs in Figma](https://www.figma.com/file/BJ95AjraesmRCWsKA013GS/Primer-Brand?node-id=10%3A38)

## Contributor checklist:


## Reviewer checklist:


## Screenshots:

> Please try to provide before and after screenshots or videos

###  🆕 Storybook playground

<table>
<tr>
<th> Before</th> <th> After </th>
</tr>
<tr>
<td valign="top">

![Current controls using text fields that were inferred from type definitions, but not easy to use as it requires knowledge of those prop types](https://user-images.githubusercontent.com/13340707/201116154-5342d916-4566-4718-aae7-80a0a5bca186.png)



 </td>
<td valign="top">

![Canvas showing secondary button with Storybook controls beneath with improved radio controls to allow easier toggling](https://user-images.githubusercontent.com/13340707/201115557-d369623b-aec6-44a4-87c0-cbf81420186b.png)


</td>
</tr>
</table>



### 🆕 Inactive (disabled) states

#### Primary

<table>
<tr>
<th> Light</th> <th> Dark </th>
</tr>
<tr>
<td valign="top">



![Primary button with dimmed appearance on a white background](https://user-images.githubusercontent.com/13340707/201115553-c5bfcd2d-faaf-4126-898c-c700fb11b6a1.png)


 </td>
<td valign="top">

![Primary button with dimmed appearance on a black background](https://user-images.githubusercontent.com/13340707/201115551-b8ed6ad0-dd59-489d-bf8f-822bffe42455.png)

</td>
</tr>
</table>

#### Secondary
<table>
<tr>
<th> Light</th> <th> Dark </th>
</tr>
<tr>
<td valign="top">



![Secondary button with dimmed appearance on a white background](https://user-images.githubusercontent.com/13340707/201115542-5e03137a-1deb-4bb3-9203-14b5c2ac5f5f.png)

 </td>
<td valign="top">


![Secondary button with dimmed appearance on a black background](https://user-images.githubusercontent.com/13340707/201115546-0876f5ff-4591-4220-a188-9a12f4b8a432.png)
</td>
</tr>
</table>

#### Subtle
<table>
<tr>
<th> Light</th> <th> Dark </th>
</tr>
<tr>
<td valign="top">

![Subtle button with dimmed appearance on a white background](https://user-images.githubusercontent.com/13340707/201115541-1539f3dc-64d5-4437-aee8-aee18a084de0.png)

 </td>
<td valign="top">

![Subtle button with dimmed appearance on a black background](https://user-images.githubusercontent.com/13340707/201115540-93d3fa28-3e51-4f07-99cb-2dfe250e1d49.png)

</td>
</tr>
</table>

### Updated focus styles




<table>
<tr>
<th> Light</th> <th> Dark </th>
</tr>
<tr>
<td valign="top">

![Focussed primary button on white background with a blue focus outline](https://user-images.githubusercontent.com/13340707/201115492-de38a11c-44ec-47ee-800c-a8cdba965aeb.png)

 </td>
<td valign="top">

![Focussed primary button on black background with a blue focus outline](https://user-images.githubusercontent.com/13340707/201115538-082c4d8b-445a-47e2-8c3d-8e9bbd3f0e6e.png)

</td>
</tr>
</table>
